### PR TITLE
Experimental: added LTOR query expansion technique in rephrase_using_translation 

### PR DESCRIPTION
**Pull request by karthikRavichandran**
- [x] **PR title**: "Experimental: added LTOR query expansion technique in rephrase_using_translation "
- [x] **PR message**: 
    - **Description:** Adding new query expansion technique in retriever that uses multilevel translation  
    - **Issue:** the issue # Please refer https://arxiv.org/abs/2402.13500v1  for method and initial results
    - **Dependencies:** Users have to install torch and transformers while using this functionality 
    - **Twitter handle:** [@KARTHIKRAVI007](https://twitter.com/KARTHIKRAVI007)
- [x] **Add tests and docs**: 
  1. The simple notebook is placed in `docs/docs/integrations/retrievers/rephrase_using_translation.ipynb`

### CODE REVIEW

**Comment by karthikRavichandran** [comment]
Hi @baskaryan, @efriis, @eyurtsev, @hwchase17, Can any of you please look at my pull request? I don't know why only Vercel succeeded and the other three have been running since the creation of pull request.

**Comment by hwchase17** [comment]
this is really interesting, but implementing it as a retriever is not quite the right way

I would take the logic inside rephrase_using_translation, add it to langchain-experimental as a standalone class, and then show how to use (the notebook you added is fine)

File: libs/langchain/langchain/retrievers/ltor_retriever.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from langchain_community.retrievers.ltor_retriever import (
```
[the remark this answers is not in the document]
Comment (karthikRavichandran) [reply libs/langchain/langchain/retrievers/ltor_retriever.py:1]: Sure. I will remove these imports. Also, I'll update the main class and refactore it as you mentioned in the previous comment.

**Comment by karthikRavichandran** [comment]
@hwchase17 I made the suggested changes: added a directory rephrase_using_translation in the experimental and added the file rephrase_using_translation.py which has LTORQueryExpansion class. I have also made changes to the Python Notebook. Please go review my changes and let me know if I have to modify the structure or name of the class further.

**Comment by karthikRavichandran** [comment]
@hwchase17 I made the suggested changes but a few pylint tests are failing. I couldn't understand this error "I001 Import block is un-sorted or un-formatted" for ___init__.py file, as this file has only one import.

**Comment by baskaryan** [comment]
> @hwchase17 I made the suggested changes but a few pylint tests are failing. I couldn't understand this error "I001 Import block is un-sorted or un-formatted" for _**init**.py file, as this file has only one import.

to format and lint locally run
```bash
cd libs/experimental
poetry install --with lint,test,typing
make format
make lint
```

**Comment by karthikRavichandran** [comment]
> > @hwchase17 I made the suggested changes but a few pylint tests are failing. I couldn't understand this error "I001 Import block is un-sorted or un-formatted" for _**init**.py file, as this file has only one import.
> 
> to format and lint locally run
> 
> ```shell
> cd libs/experimental
> poetry install --with lint,test,typing
> make format
> make lint
> ```
@baskaryan 
Thanks for the help. I have made the respective changes and pushed the code. Here, after running `make format`, it formatted the __init__.py, but the changes it made still exceeded the 88 char in a line. It seems there is no solution for this, please check this (https://github.com/astral-sh/ruff/issues/4894). So, I renamed the file to keep the import (at  __init__.py) in Pylint's 88  char line constraint.

**Comment by karthikRavichandran** [comment]
@hwchase17 @baskaryan . I've made suggested chages and verfied the testcase in local (pylint). Please let me if I need to change anything.

**Comment by karthikRavichandran** [comment]
@baskaryan @hwchase17 I have fixed all CI errors, now all the checks are 💯  done and successful . Can you please verify it and let me know the further changes.

**Comment by karthikRavichandran** [comment]
@hwchase17 @baskaryan , please let me know if you want me to change anything in the code- like class name or function structure. All the test cases are passing as of now

**Comment by karthikRavichandran** [comment]
Hi @baskaryan, @efriis, @eyurtsev, @hwchase17, Can any of you please look at my pull request? I get success mail for all the test case but the workflows says it's till waiting for approval

**Comment by karthikRavichandran** [comment]
@baskaryan Do I have to update the PR whenever the master branch changes ? Or Do i have to wait for your review and keep the changes untouched ?.

**Comment by karthikRavichandran** [comment]
@baskaryan @hwchase17 Can you please guide me regrading this PR. Do I have to do fix or change anything in this implimentaion?

**Comment by efriis** [comment]
closing and feel free to reopen against the langchain-experimental repo (this package moved)! https://github.com/langchain-ai/langchain-experimental

sorry for the delay on this one. This doesn't really fit any of our abstractions well, and thank you for making it a standalone class! I think the remaining issue is that it still mostly implements the retriever interface, and I'm a bit confused why it mostly deals with `Documents` as opposed to a message or strings. But I might be misunderstanding!

**Comment by efriis** [comment]
also if you transfer to new repo - please document in the docstring so it shows up in api docs! I don't think adding a notebook to the retrievers integrations section makes sense for this (will show in api docs though)

**Pull request closed without merging**